tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1

```diff
@@ -182,47 +262,71 @@ if ($retval) {
+    $errOut = $($retVal = & {az group create --location $Location --resource-group $ResourceGroupName --subscription $SubscriptionId}) 2>&1
+    if ($LASTEXITCODE -ne 0)
+    {
+        throw "Error creating ResourceGroup ($ResourceGroupName): $errOut"
+    }
 } 
 
 # Register the required resource providers 
-Write-Host "Registering the required resource providers for AIO" -ForegroundColor Cyan
-az provider register -n "Microsoft.ExtendedLocation"
-az provider register -n "Microsoft.Kubernetes"
-az provider register -n "Microsoft.KubernetesConfiguration"
-az provider register -n "Microsoft.IoTOperationsOrchestrator"
-az provider register -n "Microsoft.IoTOperationsMQ"
-az provider register -n "Microsoft.IoTOperationsDataProcessor"
-az provider register -n "Microsoft.DeviceRegistry"
+$resourceProviders = 
+@(
+    "Microsoft.ExtendedLocation",
+    "Microsoft.Kubernetes",
+    "Microsoft.KubernetesConfiguration"
+)
+foreach($rp in $resourceProviders)
+{
+    $errOut = $($obj = & {az provider show -n $rp | ConvertFrom-Json}) 2>&1
+    if ($LASTEXITCODE -ne 0)
+    {
+        throw "Error querying provider $rp : $errOut"
+    }
 
-# Arc-enable the Kubernetes cluster
-Write-Host "Arc enable the kubernetes cluster $ClusterName" -ForegroundColor Cyan
+    if ($obj.registrationState -eq "Registered")
+    {
+        continue
+    }
 
-$tags = @("SKU=AKSEdgeEssentials")
-$modVersion = (Get-Module AksEdge).Version
-if ($modVersion) { 
-    $tags += @("Version=$modVersion") 
-}
-$infra = Get-AideInfra
-if ($infra) { 
-    $tags += @("Infra=$infra") 
-}
-$clusterid = $(kubectl get configmap -n aksedge aksedge -o jsonpath="{.data.clustername}")
-if ($clusterid) { 
-    $tags += @("ClusterId=$clusterid") 
+    $errOut = $($retVal = & {az provider register -n $rp}) 2>&1
+    if ($LASTEXITCODE -ne 0)
+    {
+        throw "Error registering provider $rp : $errOut"
+    }
 }
-az connectedk8s connect -n $ClusterName -l $Location -g $ResourceGroupName --subscription $SubscriptionId --tags $tags | Out-Null
+
+# Arc-enable the Kubernetes cluster
+Write-Host "Arc enable the kubernetes cluster $ClusterName" -ForegroundColor Cyan
+New-ConnectedCluster -clusterName $ClusterName -arcArgs $aideuserConfigJson.Azure -useK8s:$UseK8s
 
 # Enable custom location support on your cluster using az connectedk8s enable-features command
 Write-Host "Associate Custom location with $ClusterName cluster"
-$objectId = az ad sp show --id bc313c14-388c-4e7d-a58e-70017303ee3b --query id -o tsv
-az connectedk8s enable-features -n $ClusterName -g $ResourceGroupName --custom-locations-oid $objectId --features cluster-connect custom-locations | Out-Null
+$objectId = $aideuserConfigJson.Azure.CustomLocationOID
+if ([string]::IsNullOrEmpty($objectId))
+{
+    $customLocationsAppId = "bc313c14-388c-4e7d-a58e-70017303ee3b"
+    $errOut = $($objectId = & {az ad sp show --id $customLocationsAppId --query id -o tsv}) 2>&1
```

Comment:
i think this requires owner role to query this oid.

Reply:
not just Owner role. It requires Read.All & ReadWrite.All permissions on Graph API...

Reply:
those are assumed to be done already per AIO team (IT role). now we also allow passing CustomLocationId...